### PR TITLE
Namespaces > action Sign all collections

### PR DESCRIPTION
**Pull request by acruzgon**
This PR addresses [AAH-2474](https://issues.redhat.com/browse/AAH-2474)

### CODE REVIEW

File: frontend/hub/namespaces/hooks/useHubNamespaceActions.tsx [hunk 1]
```diff
@@ -11,18 +11,30 @@ import {
...
+  isDetailsPageAction?: boolean;
 }) {
   if (!options) {
-    options = { onHubNamespacesDeleted: () => {} };
+    options = { onHubNamespacesDeleted: () => {}, onHubNamespacesSignAllCollections: () => {} };
   }
   const { t } = useTranslation();
+  const { settings, featureFlags } = useHubContext();
+  const signing_service = settings.GALAXY_COLLECTION_SIGNING_SERVICE;
+  const can_upload_signatures = featureFlags.can_upload_signatures;
```
Comment (ZitaNemeckova) [inline frontend/hub/namespaces/hooks/useHubNamespaceActions.tsx:29]: This is related to Insights mode. cc @himdel for details

**Review by ZitaNemeckova — DISMISSED** [review]

File: frontend/hub/namespaces/hooks/useHubNamespaceActions.tsx (hunk 1 above)
Comment (himdel) [reply frontend/hub/namespaces/hooks/useHubNamespaceActions.tsx:29]: It should be `can_create_signatures`, as in https://github.com/ansible/ansible-hub-ui/blob/master/src/utilities/can-sign.ts#L7 :)

Comment (himdel) [reply frontend/hub/namespaces/hooks/useHubNamespaceActions.tsx:29]: Ooh... the original is `canSign && !can_upload_signatures`. So `!can_upload_signatures` is right :+1:
It's just missing that `canSign` part, which boils to that `can_create_signatures` (+ permissions)

File: frontend/hub/collections/hooks/useSignAllCollections.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            <SplitItem />
+            <SplitItem>
+              <Trans>Unsigned version(s)</Trans>
+            </SplitItem>
+          </Split>
+        </GridItem>
+        <GridItem span={12}>
+          <FormGroup fieldId="service-selector" label={t`Signing service selector:`}>
+            <FormSelect value="ansible-default" id="service-selector">
+              <FormSelectOption value="ansible-default" label={t('ansible-default')} />
```
Comment (himdel) [inline frontend/hub/collections/hooks/useSignAllCollections.tsx:110]:
```suggestion
              <FormSelectOption value="ansible-default" label={'ansible-default'} />
```

File: frontend/hub/namespaces/hooks/useHubNamespaceActions.tsx [hunk 3]
```diff
@@ -44,6 +56,22 @@ export function useHubNamespaceActions(options?: {
...
+        icon: KeyIcon,
+        label: t(`Sign all collections`),
+        onClick: (namespace) =>
+          signCollection({
+            onComplete: options.onHubNamespacesSignAllCollections,
+            namespace: namespace,
+            signing_service: signing_service ?? '',
+          }),
+        isDisabled: () => (canSignAllCollections() ? '' : t('Select a repository filter')),
+        isHidden: () => !options.isDetailsPageAction,
```
Comment (himdel) [inline frontend/hub/namespaces/hooks/useHubNamespaceActions.tsx:72]:
```suggestion
        isHidden: () => !options.isDetailsPageAction || !can_create_signatures,
```

Signing is still configurable, so we need these hidden when signing is off.

Comment (MilanPospisil) [reply frontend/hub/namespaces/hooks/useHubNamespaceActions.tsx:72]: However, I have it as: isHidden: () => !(options.isDetailsPageAction && canSignAllCollections()), which is: 

const canSignAllCollections = () => can_create_signatures && !can_upload_signatures;

Is this correct?

Comment (himdel) [reply frontend/hub/namespaces/hooks/useHubNamespaceActions.tsx:72]: yup, either `!options.isDetailsPageAction || !canSignAllCollections()` or `!(options.isDetailsPageAction && canSignAllCollections())` both make sense :+1:

**Review by ZitaNemeckova — APPROVED** [review]

**Review by MilanPospisil — APPROVED** [review]
LGTM :+1:

**Pull request merged**